File: android/app/src/main/java/com/smartexpense/android/domain/usecase/notification/MarkNotificationReadUseCase.java
package com.smartexpense.android.domain.usecase.notification;

import com.smartexpense.android.domain.repository.NotificationRepository;
import com.smartexpense.android.domain.repository.ResultCallback;

public class MarkNotificationReadUseCase {
    private final NotificationRepository repository;

    public MarkNotificationReadUseCase(NotificationRepository repository) {
        this.repository = repository;
    }

    public void execute(String id, ResultCallback<Void> callback) {
        repository.markAsRead(id, callback);
    }
}

File: android/app/src/main/kotlin/com/smartexpense/android/domain/usecase/notification/MarkNotificationReadUseCase.kt
package com.smartexpense.android.domain.usecase.notification

import com.smartexpense.android.domain.repository.NotificationRepository

class MarkNotificationReadUseCase(private val repository: NotificationRepository) {
    suspend fun execute(id: String): Result<Unit> {
        return repository.markAsRead(id)
    }
}
